recipe/meta.yaml

```diff
@@ -25,7 +25,7 @@ requirements:
     - importlib-resources  # Normally only needed for py<39, but we're going noarch here!
     - importlib-metadata  # Normally only needed for py<38, but we're going noarch here!
     - mne-base >=1.0
-    - numpy >=1.16.0
+    - numpy >=1.21
```

Comment:
1.21.0 is only ~1 year old (with the latest patch version coming out in April 22), it seems a bit stricter than most projects. How much does this high minimum requirement gain you? In MNE at least if we need something in a very recent version, we add them to `.fixes` until our minimum requirement goes up (we follow the ~2 year standard that other projects like scipy/pandas etc. seem to be trying to follow)

Reply:
We used type hints and `mypy` in `mne-icalabel`, especially `numpy.typing.NDArray`, added in 1.21. So if you have an older version `import mne_icalabel` raises.

Reply:
We could also remove those type-hints, or replace them with the more general `ArrayLike` introduced in 1.20.

Reply:
I'd argue that since this is a new package, it's okay to have "modern" dependencies

Type hints make it easier to avoid bugs, so users gain something substantial in exchange

Reply:
1.20 is ~1.5 years old so that seems at least better. But you can also try 1.21 and see whether or not it breaks stuff for people. I think it should at least be okay for MNE since I don't think any of our deps pin to a numpy older than this.

Reply:
If you don't mind, I think I prefer to stick with `1.21`. To my knowledge, there is no `mne` related package that pins to an older dep. I'm using regularly `1.21` to `1.23` on macOS, Linux and Windows with `mne` related packages and at least for now, nothing broke.

And I like type-hints 😉 They make it easy to know what you are working with. I'm still waiting for a nice type-hint for the array shapes 😄 

Let me know if you think this is a no-go; then I'll open a PR in `mne-icalabel` to replace the type-hints with `ArrayLike` (numpy `1.20`).

Reply:
It's okay with me, just want to make sure you're aware of the potential drawbacks!

Reply:
All right, I will keep this cycle in mind in the future :)
Thanks for the feedback, in it goes.